### ISSUE TITLE
Residual Net / ResNet in TL

### ISSUE THREAD

**Issue by geometrikal** [tensorlayer/tensorlayer#196]
I have been successfully using the resnet example code [here](https://github.com/ritchieng/wideresnet-tensorlayer). It would be nice to have a more concise form.

What do you think about adding a ResNet layer type? Interface would be similar to Conv2D, with addition of some flags such as `type` - what topology (BN-C-R / C-BN-R etc), and `subsample` - do we subsample by 2, etc.

**Comment by zsdonghao** [comment]
I am thinking about it as well~ 
But I found there are many kinds of resnet, for me, I found it is easier to build residual blocks by using a for loop. This is a open topic, please let me know, if anyone have suggestion.
 
BTW, I think this resnet example is clear than the one you read.
https://github.com/zsdonghao/SRGAN/blob/master/model.py
```python
for i in range(16):
      nn = Conv2d(n, 64, (3, 3), (1, 1), padding='SAME', name='c1%s' % i)
      nn = BatchNormLayer(nn, act=tf.nn.relu, is_train=is_train, name='b1%s' % i)
      nn = Conv2d(nn, 64, (3, 3), (1, 1), padding='SAME', name='c2%s' % i)
      nn = BatchNormLayer(nn, is_train=is_train, name='b2%s' % i)
      nn = ElementwiseLayer([n, nn], tf.add, 'add%s' % i)
      n = nn
```

**Comment by geometrikal** [comment]
Yes, that is nicely compact and clear. Maybe it is simpler to just add ResNet example (one that includes the downsampling and padding)? Off topic, but have you found that different ResNet block topologies are more suited to different types of data?

**Comment by zsdonghao** [comment]
Yes, I will summarise the examples.
I usually use the traditional one, I didn't find any big performance differences on different topologies on image applications.
